crates/sui-transactional-test-runner/src/test_adapter.rs

```diff
@@ -248,36 +248,47 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
-                SuiInitArgs {
-                    accounts,
-                    protocol_version,
-                    max_gas,
-                },
-            )) => {
-                let map = verify_and_create_named_address_mapping(named_addresses).unwrap();
-                let accounts = accounts
-                    .map(|v| v.into_iter().collect::<BTreeSet<_>>())
-                    .unwrap_or_default();
-                let mut protocol_config = if let Some(protocol_version) = protocol_version {
-                    ProtocolConfig::get_for_version(protocol_version.into(), Chain::Unknown)
-                } else {
-                    ProtocolConfig::get_for_max_version_UNSAFE()
-                };
-                if let Some(mx_tx_gas_override) = max_gas {
-                    protocol_config.set_max_tx_gas_for_testing(mx_tx_gas_override)
+        let (additional_mapping, account_names, mut protocol_config, version) =
+            match task_opt.map(|t| t.command) {
+                Some((
+                    InitCommand { named_addresses },
+                    SuiInitArgs {
+                        accounts,
+                        protocol_version,
+                        max_gas,
+                    },
+                )) => {
+                    let map = verify_and_create_named_address_mapping(named_addresses).unwrap();
+                    let accounts = accounts
+                        .map(|v| v.into_iter().collect::<BTreeSet<_>>())
+                        .unwrap_or_default();
+                    let version;
+                    let mut protocol_config = if let Some(protocol_version) = protocol_version {
+                        version = protocol_version.into();
+                        ProtocolConfig::get_for_version(protocol_version.into(), Chain::Unknown)
+                    } else {
+                        version = ProtocolVersion::MAX;
+                        ProtocolConfig::get_for_max_version_UNSAFE()
+                    };
+                    if let Some(mx_tx_gas_override) = max_gas {
+                        protocol_config.set_max_tx_gas_for_testing(mx_tx_gas_override)
+                    }
+                    (map, accounts, protocol_config, version)
                 }
-                (map, accounts, protocol_config)
-            }
-            None => (
-                BTreeMap::new(),
-                BTreeSet::new(),
-                ProtocolConfig::get_for_max_version_UNSAFE(),
-            ),
-        };
+                None => {
+                    let protocol_config = ProtocolConfig::get_for_max_version_UNSAFE();
+                    (
+                        BTreeMap::new(),
+                        BTreeSet::new(),
+                        protocol_config,
+                        ProtocolVersion::MAX,
+                    )
+                }
+            };
+
+        if version != 19.into() {
+            protocol_config.set_shared_object_deletion_for_testing();
+        }
```

Comment:
What is this for?

Reply:
this is for allowing us to keep testing existing functionality when the feature flag is still off